bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/api/SipStatus.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.doorbird.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO.SipStatusBha;
+import org.openhab.binding.doorbird.internal.model.SipStatusDTO.SipStatusBha.SipStatusArray;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link SipStatus} holds SIP status information retrieved from the Doorbell.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class SipStatus {
```

Comment:
If you added getters to the SipStatusDTO that do the same logic here, couldn't you get rid of this class? 🤔 
Copying so many fields at once makes developer mistakes very easy if fields are changed later on.
Note: This is more of a suggestion than a change requirement, so take it with a grain of salt.

Reply:
The json object returned from the doorbird is needlessly complex (multiple levels and an array that always has 1 element). That's why I tried to separate the SipStatusDTO object from the object used in the handler (SipStatus). However, I reworked things a bit to handle the nullness checks in the SipStatus getters, as well as reduce the number of elements in SipStatus to those actually used by the binding. I think this makes it a bit more readable and less error-prone.